scripts/generateForDay.sh

```diff
@@ -1,41 +1,168 @@
 #!/bin/bash
-# This script allows users to manually generate keograms,startrails, and timelapses,
-# outside of the ones generated automatically.
 
-source $ALLSKY_HOME/config.sh
-source $ALLSKY_HOME/scripts/filename.sh
+# This script allows users to manually generate or upload keograms,startrails, and timelapses.
 
-cd  $ALLSKY_HOME/scripts
+ME="$(basename "${BASH_ARGV0}")"
 
-ME="$(basename "$BASH_ARGV0")"	# Include script name in output so it's easier to find in the log file
+# Allow this script to be executed manually, which requires several variables to be set.
+if [ -z "${ALLSKY_HOME}" ] ; then
+	export ALLSKY_HOME=$(realpath $(dirname "${BASH_ARGV0}")/..)
+fi
+
+# If the first argument is "--upload" then we upload files, otherwise we create them.
+if [ "$1" = "--upload" ]; then
+	shift
+	TYPE="UPLOAD"
+	MSG1="upload"
+	MSG2="uploaded"
+else
+	TYPE="GENERATE"
+	MSG1="create"
+	MSG2="created"
+fi
+
+source "${ALLSKY_HOME}/variables.sh"
+source "${ALLSKY_CONFIG}/config.sh"
+source "${ALLSKY_SCRIPTS}/filename.sh"
+[ "${TYPE}" = "UPLOAD" ] && source "${ALLSKY_SCRIPTS}/ftp-settings.sh"
 
-RED='\033[0;31m'
-GREEN='\033[0;32m'
-NC='\033[0m' # No Color
 
-if [ $# -lt 1 ]
-  then
-    echo -en "${ME}: ${RED}You need to pass a day argument\n"
-        echo -en "    ex: ${ME} 20180119${NC}\n"
-        exit 3
+if [ $# -eq 0 -o "${1}" = "-h" -o "${1}" = "--help" ]; then
+	echo -en "${RED}"
```

Comment:
Why `${RED}`? I can see flagging an error if no args or some unreasonable flags are given, but it's not an error to ask for help.

Reply:
Now only uses red on usage error, otherwise no color.